src/components/ClusterNodes/ClusterSelectionList.tsx

```diff
@@ -46,12 +62,63 @@ const ClusterSelectionList: React.FC<ClusterSelectionType> = ({
     const history = useHistory()
     const [lastSyncTime, setLastSyncTime] = useState<Dayjs>(dayjs())
 
-    const { searchKey, handleSearch, clearFilters } = useUrlFilters()
+    const { searchKey, clusterFilter, updateSearchParams, handleSearch, clearFilters } = useUrlFilters<
+        void,
+        { clusterFilter: ClusterFiltersType }
+    >({ parseSearchParams })
 
     const filteredList = useMemo(() => {
         const loweredSearchKey = searchKey.toLowerCase()
-        return clusterOptions.filter((option) => !searchKey || option.name.toLowerCase().includes(loweredSearchKey))
-    }, [searchKey, clusterOptions])
+        return clusterOptions.filter((option) => {
+            const filterCondition =
+                clusterFilter === ClusterFiltersType.ALL_CLUSTERS ||
+                !option.status ||
+                option.status === ClusterStatusByFilter[clusterFilter]
+
+            return (!searchKey || option.name.toLowerCase().includes(loweredSearchKey)) && filterCondition
+        })
+    }, [searchKey, clusterOptions, clusterFilter])
```

Comment:
Imo, the cluster filter would be having new memory address on each re-render
```suggestion
    }, [searchKey, clusterOptions, `${clusterFilter}`])
```